pages/index.js

```diff
@@ -1,7 +1,105 @@
+import { exercises } from "@/lib/exercises";
+import Image from "next/image";
+import styled from "styled-components";
+
 export default function HomePage() {
   return (
-    <div>
-      <h1>Hello from Next.js</h1>
-    </div>
+    <>
+      <Header>
+        <h1>GymLog</h1>
+        <h2>Your ultimate Fitness platform</h2>
+      </Header>
+      <Main>
+        <ExerciseList>
+          {exercises.map((exercise) => {
+            return (
+              <ExerciseCard key={exercise.id}>
+                <ImageContainer>
+                  <StyledImage
+                    src={exercise.imageUrl}
+                    alt={exercise.name}
+                    layout="fill"
+                  />
+                </ImageContainer>
+                <ExerciseName>{exercise.name}</ExerciseName>
+                <MuscleGroups>
+                  {exercise.muscleGroups.map((muscle, index) => {
+                    return <Muscle key={index}>{muscle}</Muscle>;
+                  })}
+                </MuscleGroups>
+              </ExerciseCard>
+            );
+          })}
+        </ExerciseList>
+      </Main>
+    </>
   );
 }
+
+const Header = styled.header`
+  text-align: center;
+  margin-bottom: 20px;
+`;
+
+const Main = styled.main`
+  display: flex;
+  justify-content: center;
+`;
+
+const ExerciseList = styled.ul`
+  list-style: none;
+  padding: 0;
+  display: grid;
+  grid-template-columns: repeat(auto-fill, minmax(220px, 1fr));
+  gap: 20px;
+`;
+
+const ExerciseCard = styled.li`
+  background-color: #f0f4f8;
+  border-radius: 10px;
+  padding: 15px;
+  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
+  text-align: center;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  height: 400px;
+  width: 300px;
```

Comment:
It's best not to work with fixed width and height. You could use a responsive unit here instead, e.g. %. An `li` element could have 80% of the height/width of the `ul`. Just as an example. 🙂